project/ReflectiveCodeGen.scala

```diff
@@ -0,0 +1,71 @@
+import sbt._
+import sbt.Keys._
+
+import sbtprotoc.ProtocPlugin
+import ProtocPlugin.autoImport.PB
+import akka.grpc.sbt.AkkaGrpcPlugin
+import akka.grpc.sbt.AkkaGrpcPlugin.autoImport._
+import sbt.internal.inc.classpath.ClasspathUtilities
+
+/**
+ * A plugin that allows to use a code generator compiled in one subproject to be used in a test project
+ */
+object ReflectiveCodeGen extends AutoPlugin {
+
+  override def requires = AkkaGrpcPlugin
+  override def trigger = noTrigger
+
+  override def projectSettings =
+    Seq(
+      Compile / akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Java),
+      Test / akkaGrpcGeneratedSources := Seq(AkkaGrpc.Client),
+      Compile / javacOptions ++= Seq("-encoding", "UTF-8", "-source", "11", "-target", "11"),
+      Compile / sourceGenerators += runCodeGenTask.taskValue
+    ) ++ attachProtobufDescriptorSets
+
+  def runAkkaServerlessCodegen(classpath: Classpath, protobufDescriptor: File, genSrcDir: File): Seq[File] = {
+
+    val cp = classpath.map(_.data)
+    val loader = ClasspathUtilities.toLoader(cp, AkkaGrpcPlugin.getClass.getClassLoader)
+    import scala.reflect.runtime.universe
+    import scala.tools.reflect.ToolBox
+
+    val tb = universe.runtimeMirror(loader).mkToolBox()
+
+    val source =
+      s"""
+      |import com.lightbend.akkasls.codegen.AkkaServerlessCodegenPlugin
+      |
+      |(protobufDescriptor: java.io.File, genSrcDir: java.io.File) => AkkaServerlessCodegenPlugin.runAkkaServerlessCodegen(protobufDescriptor, genSrcDir)
```

Comment:
I'm now wondering if we can do the same but calling the generator directly and without an intermediate plugin.

Reply:
yep, it works. I removed in the intermediate plugin